### PR TITLE
feat: translate bills in a single web page by wasm

### PR DESCRIPTION
**Pull request by Triple-Z**
Signed-off-by: TripleZ <me@triplez.cn>

### CODE REVIEW

**Comment by Triple-Z** [comment]
TODO:
- [ ] build constraints for `syscall/js`.
- [ ] index page read input/config from their textarea.

**Comment by Triple-Z** [comment]
@gaocegege now you can review all the go code under the `pkg` directory.

**Comment by Triple-Z** [comment]
Known issues: 
- Error: total length of command line and environment variables exceeds limit
  Related links: 
    - https://github.com/golang/go/issues/49011
    - https://groups.google.com/g/golang-codereviews/c/-N-Smb0Cfxc

File: pkg/cmd/translate.go [hunk 1]
```diff
@@ -17,12 +17,10 @@ limitations under the License.
...
 
 import (
-	"fmt"
-	"log"
-	"os"
-
+	"github.com/deb-sig/double-entry-generator/pkg/cmd/validator"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"log"
```
Comment (gaocegege) [inline pkg/cmd/translate.go:23]: Please keep log in the first import group.

File: pkg/config/init_config_wasm.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+//go:build js && wasm
+// +build js,wasm
+
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/spf13/viper"
```
Comment (gaocegege) [inline pkg/config/init_config_wasm.go:9]: Please keep this package in the second import group.

File: pkg/io/reader/file_reader.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+//go:build !js
+// +build !js
+
+package reader
+
+import (
+	"golang.org/x/text/encoding/simplifiedchinese"
+	"golang.org/x/text/transform"
+	"io"
+	"os"
```
Comment (gaocegege) [inline pkg/io/reader/file_reader.go:10]: Same here.

File: pkg/provider/alipay/alipay.go [hunk 4]
```diff
@@ -17,17 +17,12 @@ limitations under the License.
 package alipay
 
 import (
-	"bufio"
 	"encoding/csv"
 	"fmt"
+	"github.com/deb-sig/double-entry-generator/pkg/io/reader"
+	"github.com/deb-sig/double-entry-generator/pkg/ir"
```
Comment (gaocegege) [inline pkg/provider/alipay/alipay.go:23]: This should be in a separate import group

File: pkg/cmd/translate.go (hunk 1 above)
Comment (Triple-Z) [reply pkg/cmd/translate.go:23]: `gofmt` ?

Comment (gaocegege) [reply pkg/cmd/translate.go:23]: Yep

Comment (Triple-Z) [reply pkg/cmd/translate.go]: Fixed.

File: pkg/config/init_config_wasm.go (hunk 2 above)
Comment (Triple-Z) [reply pkg/config/init_config_wasm.go:10]: Fixed.

File: pkg/io/reader/file_reader.go (hunk 3 above)
Comment (Triple-Z) [reply pkg/io/reader/file_reader.go:8]: Fixed.

**Review by gaocegege — APPROVED** [review]

**Pull request merged**